src/lib/AnnotationModule.ts

```diff
@@ -0,0 +1,50 @@
+import { ANNOTATION_COLOR_KEY } from './constants';
+import Cache from './Cache';
+
+export enum AnnotationColor {
+    BLUE = '#0061d5',
+    GREEN_LIGHT = '#26c281',
+    GRIMACE = '#4826c2',
+    WATERMELON_RED = '#ed3757',
+    YELLORANGE = '#f5b31b',
+    YELLOW = '#ffd700',
+}
+
+type Subscription = (state: AnnotationColor) => void;
+
+export default class AnnotationModule {
+    private cache: Cache;
+
+    private subscriptions: Subscription[] = [];
+
+    constructor(cache = new Cache()) {
+        this.cache = cache;
+
+        if (!this.cache.get(ANNOTATION_COLOR_KEY)) {
+            this.cache.set(ANNOTATION_COLOR_KEY, AnnotationColor.BLUE, true);
+        }
+    }
+
+    public getColor = (): AnnotationColor => this.cache.get(ANNOTATION_COLOR_KEY) as AnnotationColor;
+
+    public transition = (color: AnnotationColor): AnnotationColor => {
```

Comment:
This seems to perpetuate the issue we have with AnnotationsFSM, in that it isn't extensible. It also duplicates quite a bit of logic around subscriptions. Is this much complexity needed for one property? If so, I think we'll want to consider a more holistic solution. Should migrate to redux?

Reply:
I think we may not need to migrate to redux. Instead, for this case, whenever the user updates the current selected color, we can just call `annotator.emit` to propagate the latest color down to annotations repo.

Reply:
We no longer need the subscribe / transition logic if we use the `annotator.emit` approach.

Reply:
As for the `annotator.emit` approach, I think that will be done in a separate PR (it makes more sense to add it when we have the drawing logic already merged so that we can test that when we update the color, we do see that a new drawing is using that updated color).